### ISSUE TITLE
Remove a record from the group by drag and drop it to the group

### ISSUE THREAD

**Issue by linsui** [JabRef/jabref#9189]
**Describe the solution you'd like**
A record can be added to a group by dragging and dropping it. It's natual to drag and drop it again to remove the record from the group.

**Comment by AEgit** [comment]
I disagree - so far JabRef's implementation has always been, that dragging an item to a group adds it to the group. But it does NOT remove it. I. e. you can drag'n'drop items to various groups and the respective item will be added to those groups. But you cannot remove items from groups by drag'n'dropping them. And that is a good implementation, I feel. So, I would argue to keep the current approach and to NOT change it.

**Comment by linsui** [comment]
I'm not sure what your point is. You can still drag an item to a group to add it.

**Comment by AEgit** [comment]
Ok, maybe I misunderstand you. What implementation do you have in mind that allows at the same time an item to be added to multiple groups but also removes an item from a group? The only way I could see that implemented would be to have an additional context menu and/or a switch in general settings that either allows items to be dragged and added to multiple groups OR one that only allows an item to be dragged to a single group, i. e. if it is dragged from its current parent group it is added to the new parent group and removed from the old one.

**Comment by linsui** [comment]
What is the current behavior? I don't know how to drag an item to multiple groups...

**Comment by AEgit** [comment]
You just pick an item and drag and drop it onto one group. Then you pick the same item and drag'n'drop it onto another group. Repeat for as many groups as you like.

**Comment by linsui** [comment]
Ah, got it. What I want is:
When I drag an item to a group, if it's already in that group then remove it from the group. If it's not there, add it.

**Comment by AEgit** [comment]
I see - hmm, that could work, though I am not sure, whether this is what most users would expect. Currently if you drag an item to a group it is already part of, nothing happens (or rather, it stays with the group). This new change would lead to a behaviour where in certain cases, dragging an item to a group actually removes it.

**Comment by yanyanliu1400** [comment]
Hi, I am a new contributor and I would like to make an attempt at resolving this issue. Could it please be assigned to me?

I have followed [the instructions](https://devdocs.jabref.org/getting-into-the-code/guidelines-for-setting-up-a-local-workspace.html) and set up a local workspace.

I have also read the [contribution guide](https://devdocs.jabref.org/contributing).

I was hoping I could receive some advice/pointers on where to begin with this issue.

**Comment by mlep** [comment]
@yanyanliu1400 Welcome to JabRef!
This issue is still under discussion, and a final decision has not been made (to implement it or not? An alternative? etc.). So, I suggest you choose among the (many) other issues. As a new contributor, you may prefer the ones tagged as "good first issue" (https://github.com/JabRef/jabref/issues?q=is%3Aissue+is%3Aopen+label%3A%22good+first+issue%22).

**Comment by ThiloteE** [comment]
i will increase priority to normal, once discussion has finished

**Comment by mlep** [comment]
@ThiloteE: Wouldn't a tag "Ready for implementation" be useful?

**Comment by Siedlerchr** [comment]
I suggest the other way around. Labeling issues that still need a decision.

**Comment by mlep** [comment]
I was thinking of adding a trash icon to remove a record from a group by drag-and-drop. However, I am not sure that the consequences will be clear to a user: will it remove the record from the selected groups? from all the groups? And if the selected group is the main group ("all entries"), the action should do nothing (or delete the entry from the database?). Difficult to get a IU with a clear purpose...

**Comment by mlep** [comment]
@Siedlerchr Better idea! That would mean that a new entry gets this label by default, I guess.

**Comment by ThiloteE** [comment]
In theory, labeling all issues with "still requires decision" sounds good, but in practice what will happen is this:

Someone has to go through all issues to un-label them. unlabeled issues are "stuck" and "frozen" in their form and nobody will do work on them. As soon as new discussion pops up, it has to be labeled again, which requires constant attention by maintainers and constant thinking about: is this now done or can the current decision be improved upon? **Basically there is more work to do than simply giving attention to newcomers that want to work on an issue that is still in its discussion phase and notifying them about alternatives.**

Another thing to think about: Since users without elevated rights cannot assign labels, the decision to implement something would all fall back to maintainers. Not sure if this is all too desirable.

**Comment by mlep** [comment]
@ThiloteE Good point. Let's continue as it is now.

**Comment by ThiloteE** [comment]
> When I drag an item to a group, if it's already in that group then remove it from the group. If it's not there, add it.

To be honest, I think this not a good implementation. The same action will lead to two different outcomes, depending on circumstances, which will surprise users that are new to JabRef. It is very "hidden" and not self explanatory.

A clear implementation would be, if the same action will always lead to the same outcome.

**Comment by DvP17** [comment]
@ThiloteE Another way might be to do it with a key binding: E.g., holding down Ctrl while drag and drop would move the entry (adding to the new group and deleting it in the old group). If it is already in multiple groups maybe a warning could pop-up if one would want to remove multiple groups from the entry.

**Comment by ThiloteE** [comment]
some kind of keybinding is a good idea!
